logging.go

```diff
@@ -0,0 +1,27 @@
+package sgreplicate
+
+import (
+	"strings"
+	"github.com/couchbase/clog"
+	"fmt"
+)
+
+func LogTo(replication SGReplication, key string, format string, args ...interface{}) {
+	clog.To(key, prefixWithReplicationId(replication, format), args)
+}
+
+func Warn(replication SGReplication, args ...interface{}) {
+	if clog.Level <= clog.LevelWarning {
+		clog.Warn(prefixWithReplicationId(replication, fmt.Sprint(args...)))
+	}
+}
+
+func prefixWithReplicationId(replication SGReplication, raw string) string {
+	replicationId := replication.GetParameters().ReplicationId
+	if replicationId != "" {
+		return strings.Join([]string{"[",replicationId ,"] ", raw}, "")
+	}
+
+	return raw
+}
+
```

Comment:
It feels like these would be more elegant as methods on SGReplication, instead of a separate standalone methods.  It also removes the ambiguity around LogTo and Warn.